src/sentry/snuba/entity_subscription.py

```diff
@@ -0,0 +1,274 @@
+        # rate alerts that is used to identify if we are below a general minimum alert threshold
+        count_col = re.search(r"(sessions|users)", self.aggregate)
+        if not count_col:
+            raise UnsupportedQuerySubscription(
+                "Only crash free percentage queries are supported for subscriptions"
+                "over the sessions dataset"
+            )
+        count_col_matched = count_col.group()
+
+        aggregations += [f"identity({count_col_matched}) AS {CRASH_RATE_ALERT_SESSION_COUNT_ALIAS}"]
+        functions_acl = ["identity"]
+        snuba_filter = get_filter(query, params=params)
+        snuba_filter.update_with(
+            resolve_field_list(
+                aggregations, snuba_filter, auto_fields=False, functions_acl=functions_acl
+            )
+        )
+        snuba_filter = resolve_snuba_aliases(snuba_filter, resolve_func)[0]
+        if environment:
+            snuba_filter.conditions.append(["environment", "=", environment.name])
+        return snuba_filter
+
+    def get_entity_extra_params(self) -> Mapping[str, Any]:
+        return {"organization": self.org_id}
+
+
+class MetricsCountersEntitySubscription(BaseEntitySubscription):
+    dataset = Dataset.Metrics.value
+    entity_key = EntityKey.MetricsCounters.value
+
+    def __init__(self, aggregate: str, extra_fields: Optional[Mapping[str, Any]] = None):
+        super().__init__(aggregate, extra_fields)
+        self.aggregate = aggregate
+        if not extra_fields or "org_id" not in extra_fields:
+            raise InvalidQuerySubscription(
+                "org_id is a required param when "
+                "building snuba filter for a metrics subscription"
+            )
+        self.org_id = extra_fields["org_id"]
+
+    def get_query_groupby(self) -> List[str]:
+        session_status = tag_key(self.org_id, "session.status")
+        return ["project_id", session_status]
+
+    def build_snuba_filter(
+        self,
+        query: str,
+        environment: Optional[Environment],
+        params: Optional[Mapping[str, Any]] = None,
+    ) -> Filter:
+        snuba_filter = get_filter(query, params=params)
+        conditions = copy(snuba_filter.conditions)
+        snuba_filter.update_with(
+            {
+                "aggregations": [["sum(value)", None, "value"]],
+                "conditions": conditions + [["metric_id", "=", metric_id(self.org_id, "session")]],
+                "groupby": self.get_query_groupby(),
+            }
+        )
+        if environment:
+            snuba_filter.conditions.append(["environment", "=", environment.name])
+        return snuba_filter
+
+    def get_entity_extra_params(self) -> Mapping[str, Any]:
+        return {"organization": self.org_id, "groupby": self.get_query_groupby()}
+
+
+def map_aggregate_to_entity_subscription(
+    dataset: QueryDatasets, aggregate: str, extra_fields: Optional[Mapping[str, Any]] = None
+) -> BaseEntitySubscription:
```

Comment:
So I am not a fan of adding these fields because they are entity specific for example `event_types` only really applies to events and transactions. Similarly I don't really need `org_id` for events or transactions entities 
How about if I typed the `extra_fields` better ?

Reply:
Yeah, that makes sense to me. As we add typing I'm becoming less of a fan of the python `extra_fields` pattern, since it makes it harder to understand what can be passed to a function without reading code. If this is typed better it could make sense.

Another option that is probably overkill is to have a `EntityParams` dataclass that we pass through. So an example might be
```
@dataclass
class EntityParams:
    @property 
    def extra_fields(self):
        # Builds extra fields to be passed to the snuba api
        return {}

@dataclass
class MetricEntityParams(EntityParams):
    org_id: int

    @property
    def extra_fields(self):
        return {'org_id': self.org_id}
```

Not sure if I love the pattern but just an option, I'll leave it up to you either way.